apps/chat-e2e/src/tests/chatSwitch.test.ts

```diff
@@ -0,0 +1,147 @@
+        await localStorageManager.setSelectedConversation(
+          preReplayConversation,
+          comparedConversation,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Verify another conversation is selectable during being in compare mode',
+      async () => {
+        await dialHomePage.openHomePage();
+        await dialHomePage.waitForPageLoaded();
+
+        await conversationAssertion.assertConversationCursor(
+          firstConversation.name,
+          Cursors.pointer,
+        );
+        await conversations.getEntityByName(firstConversation.name).click();
+        await conversationAssertion.assertSelectedConversation(
+          firstConversation.name,
+        );
+      },
+    );
+
+    await dialTest.step('Send request to the first conversation', async () => {
+      await dialHomePage.throttleAPIResponse(API.chatHost, 60000);
+      await chat.sendRequestWithButton(request, false);
+      firstConversation.name = request;
+    });
+
+    await dialTest.step(
+      'Verify any conversation cursor is "not-allowed" during text generation',
+      async () => {
+        await conversationAssertion.assertConversationCursor(
+          preReplayConversation.name,
+          Cursors.notAllowed,
+        );
+        await conversationAssertion.assertConversationCursor(
+          firstConversation.name,
+          Cursors.notAllowed,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Verify another conversation is not selectable during text generation',
+      async () => {
+        await conversations.getEntityByName(preReplayConversation.name).click();
+        await conversationAssertion.assertSelectedConversation(
+          firstConversation.name,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Select [Replay] conversation and start generation',
+      async () => {
+        await sendMessage.stopGenerating.waitForState({ state: 'visible' });
+        await sendMessage.stopGenerating.click();
+        await conversations.getEntityByName(replayConversation.name).click();
+        await chat.startReplay();
+      },
+    );
+
+    await dialTest.step(
+      'Verify conversation cursor is "not-allowed" during the chat replay',
+      async () => {
+        await conversationAssertion.assertConversationCursor(
+          preReplayConversation.name,
+          Cursors.notAllowed,
+        );
+        await conversationAssertion.assertConversationCursor(
+          replayConversation.name,
+          Cursors.notAllowed,
+        );
+      },
+    );
+
+    await dialTest.step(
+      'Verify another conversation is not selectable during text generation in Replay',
+      async () => {
+        await conversations.getEntityByName(preReplayConversation.name).click();
+        await conversationAssertion.assertSelectedConversation(
+          replayConversation.name,
+        );
+        await sendMessage.stopGenerating.waitForState({ state: 'visible' });
```

Comment:
it is not necessary to stop conversation after all verifications

Reply:
done